tests/ui/const-generics/issues/cg-in-dyn-issue-128176.rs

```diff
@@ -1,6 +1,7 @@
 //@ check-pass
 
-// Regression test for #128176.
+// Regression test for #128176. Previously we would call `type_of` on the `1` anon const
```

Comment:
This test being an example of the hir-based logic being wrong resulting in an ICE, which was fixed by no longer calling `type_of` before the anon const had been lowered